Admin/AdminHelper.php

```diff
@@ -335,6 +334,8 @@ private function pathExists(PropertyAccessorInterface $propertyAccessor, $entity
             return true;
         } catch (NoSuchPropertyException $e) {
             return false;
+        } catch (UnexpectedTypeException $e) {
```

Comment:
When the adminHelper is trying to guess the correct path given a string it could end calling the method getValue of propertyAccessor passing as a parameter a thing different to an object or array. This is only for sf2.3.

It happened on my test and that's why it was failing on the compatibility test. I could instead change the test and remove that catch. But I thought it was more robust this way.